### DISCUSSION TITLE
ShawBluecurveWesternCanada HTML Remote.

### DISCUSSION THREAD

**Discussion by blasco403** [GerryDazoo/Slinger#329] [General]
Hello, 

I have edited the TIVO custom remote files (thanks pfuller88 - great work) to work with my Shaw Canada Bluecurve service that is based on Xfinity/Comcast. It includes the channel list for my subscribed service (edit .html to modify).

I have a SB 350 and 500, I use a unified config file.

sling350.html and sling500.html files need to be in the same folder you run the slinger software from.

The folders (sling350files and sling500files) also need to be in the same folder you run the slinger software from.
- I have two folders to display a different image on each of the remotes so that I can tell which remote I am connected to.

This works with the 3.08e version of the slinger software.

Hope this helps someone else. Big thanks to all who contributed to getting the HTML version of a remote working.

Enjoy and thanks.

[ShawBluecurveWesternCanadaHTMLRemote.zip](https://github.com/GerryDazoo/Slinger/files/10050169/ShawBluecurveWesternCanadaHTMLRemote.zip)

**Comment by GIBLechat** [comment]
@blasco403 Excellent adaptation.  I see you set it up for Shaw Calgary and I am amending the HTML code set up for Shaw Edmonton channels.  

A couple of questions if I may -- I note that in the html code, Table Class Numbers, there are 3 favch5 and no favch8 or favch9.  I assume that was an oversight?  It doesn't seem to affect the operation.

Secondly, Code 81 for my Bluecurve, does not bring up the Menu.  It does nothing.  Are you aware of any differences between Edmonton and Calgary implementations of the Remote?

Thanks.

**Reply by GerryDazoo** [reply]
About code 81. Verify it's in the keycodes[ ...] listed when you start your stream. You may have installed a different remote type when you configured the video source.

**Reply by blasco403** [reply]
Hello,

I am not sure why your keycodes could be different than mine. Both my 350 and 500 are configured identical and with a Comcast_X1-(XR5) remote. I know there are different versions of Shaw Bluecurve, my STB's are old as I moved to the service when it was first offered.

I have updated the remote quite a bit..

Here is the latest version.

[ShawBluecurveWesternCanadaHTMLRemote.zip](https://github.com/GerryDazoo/Slinger/files/10613920/ShawBluecurveWesternCanadaHTMLRemote.zip)

**Reply by GIBLechat** [reply]
blasco403.  Thanks.  I will use this and change the code accordingly for the Edmonton channels.  Not much work thankfully. My STBs are about the size of 2 cigarette packages placed together.  Luckily, they still accept IR codes and, I have been able to use the HDMI ports to stream the video.  Thanks again.

**Comment by GIBLechat** [comment]
Gerry, 

Thanks for responding so quickly.

81 does not show up.  It stops at 77.  But I have referenced the correct remote - Sling500.html from @blasco403 and, in reviewing the html code, it does have 2 references to 81 as a key to send. Any other thoughts.

**Reply by GerryDazoo** [reply]
The same one as I had before the base remote used by you and blasco403 are different, Or code 81 never worked in the first place.

**Reply by GIBLechat** [reply]
Thanks Gerry.  I have the same service as blasco43.  However, I might have the later version of the hardware.  Interestingly enough, every other key works correctly.  I guess I'll have to experiment.  Where does Slinger get the key codes from that it lists when the remote service starts up?

**Reply by GerryDazoo** [reply]
It reads the configured keycodes from the Slingbox. So if you have a different list then it means your video source was not configured with the same Remote type.

**Reply by blasco403** [reply]
When I was originally setting up slinger with only the text based remote originally provided, I figured out each code manually by editing and testing each code. This was before a list off all configured codes was presented in slinger as well. At least you now have a list of all excepted codes to test to see what they do.

Good luck.

**Reply by GerryDazoo** [reply]
Have a look at the readme it lists Keycodes and button names from an old Sling doc.
That's the good news, the bad news is there was no way to enforce their naming convention so some remotes key defs don't match. The only thing that I'm pretty sure of is the Power key and the digit keys.

**Reply by GIBLechat** [reply]
Thanks Gerry.  Here is the output from Slinger.  I will experiment and see what works.  I, and I am sure, the whole community, appreciates your work on this project.

Keycodes= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 50, 53, 56, 57, 59, 60, 62, 63, 64, 65]

**Reply by GIBLechat** [reply]
Sorry, working on 2 slingboxes in 2 locations and have mixed them up.  Here are the codes from my Shaw Bluecurve.  The output in the reply above is from a Cox Arrris AX013ANC and I cannot get any remote to work with that one.

Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 24, 25, 27, 28, 29, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 72, 73, 74, 75, 76, 77]

**Reply by blasco403** [reply]
Here is a keymap I have for a different version of the xfinity remote, maybe these are the correct keycodes:

	<KM K="001" n="Power on/off" c="General" s="O" i="POWER" F="F_POWER"/>
	<KM K="002" n="Power On" c="General" i="POWER_ON"/>
	<KM K="003" n="Power Off" c="General" i="POWER_OFF"/>
	<KM K="004" n="Channel Up" c="General" s="+" i="CHUP" F="F_CHUP"/>
	<KM K="005" n="Channel Down" c="General" s="-" i="CHDOWN" F="F_CHDOWN"/>
	<KM K="006" n="Volume Up" c="General" i="VOL_UP" F="F_VOL_UP"/>
	<KM K="007" n="Volume Down" c="General" i="VOL_DOWN" F="F_VOL_DOWN"/>
	<KM K="008" n="Mute" c="General" i="VOL_MUTE" F="F_VOL_MUTE"/>
	<KM K="009" n="Digit 1" c="Numeric Keypad" s="1" F="F_DIGIT_1"/>
	<KM K="010" n="Digit 2" c="Numeric Keypad" s="2" F="F_DIGIT_2"/>
	<KM K="011" n="Digit 3" c="Numeric Keypad" s="3" F="F_DIGIT_3"/>
	<KM K="012" n="Digit 4" c="Numeric Keypad" s="4" F="F_DIGIT_4"/>
	<KM K="013" n="Digit 5" c="Numeric Keypad" s="5" F="F_DIGIT_5"/>
	<KM K="014" n="Digit 6" c="Numeric Keypad" s="6" F="F_DIGIT_6"/>
	<KM K="015" n="Digit 7" c="Numeric Keypad" s="7" F="F_DIGIT_7"/>
	<KM K="016" n="Digit 8" c="Numeric Keypad" s="8" F="F_DIGIT_8"/>
	<KM K="017" n="Digit 9" c="Numeric Keypad" s="9" F="F_DIGIT_9"/>
	<KM K="018" n="Digit 0" c="Numeric Keypad" s="0" F="F_DIGIT_0"/>
	<KM K="019" n="Enter" c="Numeric Keypad" s="{enter}" i="ENTER" F="F_ENTER"/>
	<KM K="020" n="+100" c="Numeric Keypad"/>
	<KM K="021" n="Last Channel" c="General" s="Z" i="LASTCH" F="F_LASTCH"/>
	<KM K="022" n="Input" c="Misc" s="Y" i="INPUT" F="F_INPUT"/>
	<KM K="023" n="External" c="Misc" i="EXTERNAL" F="F_EXTERNAL"/>
	<KM K="024" n="Play" c="Playback" s="P" i="PLAY" F="F_PLAY"/>
	<KM K="025" n="Stop" c="Playback" s="S" i="STOP" F="F_STOP"/>
	<KM K="026" n="Pause" c="Playback" s="U" i="PAUSE" F="F_PAUSE"/>
	<KM K="027" n="Rewind" c="Playback" s="R" i="REWIND" F="F_REWIND"/>
	<KM K="028" n="Fast Forward" c="Playback" s="F" i="FFWD" F="F_FFWD"/>
	<KM K="029" n="Record" c="Playback" s="C" i="RECORD" F="F_RECORD"/>
	<KM K="030" n="Skip Forward" c="Playback" s=")" i="SKIPFWD" F="F_SKIPFWD"/>
	<KM K="031" n="Skip Backward" c="Playback" s="(" i="SKIPBACK" F="F_SKIPBACK"/>
	<KM K="032" n="Live TV" c="Menu" s="L" i="LIVETV" F="F_LIVETV"/>
	<KM K="033" n="Setup" c="Menu" s="N" i="SETUP" F="F_SETUP"/>
	<KM K="034" n="Menu" c="Menu" s="M" i="MENU" F="F_MENU"/>
	<KM K="035" n="Program Guide" c="Menu" s="G" i="PROGUIDE" F="F_PROGUIDE"/>
	<KM K="036" n="Exit" c="Navigation" s="X" i="EXIT" F="F_EXIT"/>
	<KM K="037" n="Clear" c="General" s="Q" i="CLEAR" F="F_CLEAR"/>
	<KM K="038" n="Cursor Up" c="Navigation" s="{up}" i="CURSOR_UP" F="F_CURSOR_UP"/>
	<KM K="039" n="Cursor Down" c="Navigation" s="{down}" i="CURSOR_DOWN" F="F_CURSOR_DOWN"/>
	<KM K="040" n="Cursor Left" c="Navigation" s="{left}" i="CURSOR_LEFT" F="F_CURSOR_LEFT"/>
	<KM K="041" n="Cursor Right" c="Navigation" s="{right}" i="CURSOR_RIGHT" F="F_CURSOR_RIGHT"/>
	<KM K="042" n="Select/OK" c="Navigation" s="Space" i="SELECT" F="F_SELECT,F_ENTER"/>
	<KM K="043" n="Page Up" c="Navigation" s="E" i="PGUP" F="F_PGUP"/>
	<KM K="044" n="Page Down" c="Navigation" s="D" i="PGDOWN" F="F_PGDOWN"/>
	<KM K="045" n="Favorite" c="Misc" s="B" i="FAVORITE" F="F_FAVORITE"/>
	<KM K="046" n="Info" c="Menu" s="I" i="INFO" F="F_INFO"/>
	<KM K="047" n="Angle" c="Playback" s="H" i="ANGLE" F="F_ANGLE"/>
	<KM K="048" n="Subtitle" c="Playback" i="SUBTITLE"/>
	<KM K="049" n="Surround" c="Misc"/>
	<KM K="050" n="Slow" c="Misc" i="SLOW"/>
	<KM K="051" n="Eject" c="Misc" s="J" F="F_EJECT"/>
	<KM K="052" n="Random" c="Playback"/>
	<KM K="053" n="Custom01"/>
	<KM K="054" n="Custom02"/>
	<KM K="055" n="Custom03"/>
	<KM K="056" n="Custom04"/>
	<KM K="057" n="Custom05"/>
	<KM K="058" n="Custom06"/>
	<KM K="059" n="Custom07"/>
	<KM K="060" n="Custom08"/>
	<KM K="061" n="Custom09"/>
	<KM K="062" n="Custom10"/>
	<KM K="063" n="Custom11"/>
	<KM K="064" n="Custom12"/>
	<KM K="065" n="Custom13"/>
	<KM K="066" n="Custom14"/>
	<KM K="067" n="Custom15"/>
	<KM K="068" n="Custom16"/>
	<KM K="069" n="Custom17"/>
	<KM K="070" n="Custom18"/>
	<KM K="071" n="Custom19"/>
	<KM K="072" n="Instant Replay" c="Playback" s="(" i="INSTREPLAY" F="F_INSTREPLAY"/>
	<KM K="073" n="Xfinity" c="Menu" s="T" i="DVR" F="F_DVR" O="D-PAD"/>
	<KM K="074" n="Red/C" c="Teletext" s="A" i="RED" F="F_RED"/>
	<KM K="075" n="Green/D" c="Teletext" s="B" i="GREEN" F="F_GREEN"/>
	<KM K="076" n="Yellow/A" c="Teletext" s="V" i="YELLOW" F="F_YELLOW"/>
	<KM K="077" n="Blue/B" c="Teletext" i="BLUE" F="F_BLUE"/>

Good luck.

**Reply by GIBLechat** [reply]
blasco403, that was very helpful.  Code 73 entered the menu mode. As well, the pause and play commands are the same to I changed the Pause key to Pause/Play and changed the play key to instant replay (code 72), both just like the physical remote control I now have the remote working and will clean it up a bit before I contribute the code back to the community. Thanks again for all your help.

**Comment by GIBLechat** [comment]
Here are the lines of code:

<td><button class="blackimage" type="submit" onClick="sendKeys('81','menu'); return false;"><img class="tivobuttonimage" src="/sling500files/tivobutton.png" alt="tivo"></button></td>

<td><button class="darkblue" type="submit" onClick="sendKeys('81','dvr'); return false;">MENU</button></td>

**Reply by GIBLechat** [reply]
OOPs, when I posted, it ran the code.

I am not sure how to post without executing.

**Comment by GIBLechat** [comment]
I have now completed my final review and present for the community, Shaw BlueCurve XR11 Western Canada (Edmonton) SB500 HTML Remote Control.  I hope it works as well for those who try it as it has for me. Thanks again to blasco403 and Gerry for the help.
[ShawBlueCurveWesternCanadaEdmontonHTMLXR11RemoteSB500.zip](https://github.com/GerryDazoo/Slinger/files/10718436/ShawBlueCurveWesternCanadaEdmontonHTMLXR11RemoteSB500.zip)

**Reply by GIBLechat** [reply]
I fixed an error in the code - forgot to change the Menu code in one place from 81 to 73.  Revised remote is now attached.
[ShawBlueCurveWesternCanadaEdmontonHTMLXR11RemoteSB500_v.1.01.zip](https://github.com/GerryDazoo/Slinger/files/10777076/ShawBlueCurveWesternCanadaEdmontonHTMLXR11RemoteSB500_v.1.01.zip)

**Comment by blasco403** [comment]
Updates for Slinger V4:

Make the following changes to the HTML file:

Change this line:
`<td><button class="darkblue" type="submit" onClick="sendKeys('Restart','Restart'); return false;">Reset</button></td>`

To this:
`<td><button class="darkblue" type="submit" onClick="sendKeys('-1','Restart'); return false;">Reset</button></td>`

Change this line:
`<td class="footer"><p class="footer"><h3>Status:%s</h3></p></td>`

To this:
`<td class="footer"><p class="footer">Status:%s</p></td>`

Make the following changes to the "layout.css" in the supporting files directory: (makes status display better at bottom of remote)

Change the following lines from this:

}
table.bottom {
    width:55%;
    margin:auto;
    border-collapse:collapse;
    background-color:#000000;
}
.footer td {
    height:10px;
    margin:0px;
    padding-bottom: 10px;
}
.footer p {
    margin-left:78px;
    font-size:10px;
    color:#c9adf7;
}

To this:

}
table.bottom {
    width:85%;
    margin:auto;
    border-collapse:collapse;
    background-color:#000000;
}
.footer td {
    height:20px;
    margin-right:0;
    padding-bottom: 10px;
}
.footer p {
    font-size:11px;
    margin-right:0;
    padding-bottom: 10px;
    color:#c9adf7;
}

Enjoy...

**Comment by blasco403** [comment]
Here is the latest zip file for Version 4.

[ShawBluecurveWesternCanadaHTMLRemote.zip](https://github.com/GerryDazoo/Slinger/files/10787661/ShawBluecurveWesternCanadaHTMLRemote.zip)

**Reply by GIBLechat** [reply]
Hi blasco403.  Only the files for the SB 350 included.  Have you updated the SB500 files?
